### PR TITLE
Add nix flake

### PR DESCRIPTION
**Pull request by cojosh**
Adds a nix flake and uses it setup the build environment in our github actions.

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  description = "Fountain Toolchain";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs?ref=24.05";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }: flake-utils.lib.eachDefaultSystem (system:
+      let
```
Comment (SeineEloquenz) [inline flake.nix:9]: Why 4 spaces of indentation instead of 2 here?

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        pkgs = (import nixpkgs) {
```
Comment (SeineEloquenz) [inline flake.nix:10]: The braces around import are unnecessary.

File: flake.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          inherit system;
+        };
+      in
+      with pkgs;
+      {
+          devShells.default = mkShell {
```
Comment (SeineEloquenz) [inline flake.nix:16]: This line is indented by 2 spaces too much.

**Comment by SeineEloquenz** [comment]
Apart from some formatting nitpicks on the flake, this seems fine as it is for me.

**Review by SeineEloquenz — APPROVED** [review]
Apart from some formatting nitpicks on the flake, this seems fine as it is for me.

**Comment by cojosh** [comment]
> Apart from some formatting nitpicks on the flake, this seems fine as it is for me.

Ran nixkpgs-fmt on the flake should all be good now.

**Pull request merged**